--- playerlib.php.ori	2021-10-16 14:54:14.393819138 +0200
+++ playerlib.php	2021-10-16 15:05:58.014984137 +0200
@@ -4000,7 +4000,23 @@
 			}
 			// Song file
 			else {
-				$current['coverurl'] = '/coverart.php/' . rawurlencode($song['file']);
+        //
+                                // Here is the modification for moOdeCDPlayer
+                                //
+                                // orginal line :
+                                //   $current['coverurl'] = '/coverart.php/' . rawurlencode($song['file']);
+
+
+
+                                // CD Audio
+                                if (substr($song['file'], 0, 8) == 'cdda:///') {
+                                      $current['coverurl'] = 'imagesw/current_cd/' . basename(glob("/var/local/www/imagesw/current_cd/*.jpg")[0]);
+                                   }
+                                // Song file
+                                else {
+                                      $current['coverurl'] = '/coverart.php/' . rawurlencode($song['file']);
+                                     }
+                                //  END of the changes
 			}
 
 			if (substr($song['file'], 0, 4) == 'http') {
